src/ex_docmd.c

```diff
@@ -8928,6 +8928,10 @@ ex_cd(exarg_T *eap)
 	    /* Echo the new current directory if the command was typed. */
 	    if (KeyTyped || p_verbose >= 5)
 		ex_pwd(eap);
+
+#ifdef FEAT_AUTOCMD
+	    apply_autocmds(EVENT_DIRCHANGED, new_dir, new_dir, FALSE, curbuf);
```

Comment:
The event "pattern" should say "global" or "window". `expand('<afile>')` can be used to get the directory name itself (though even that seems useless, because `getcwd()` should serve that purpose anyway).

    apply_autocmds(EVENT_DIRCHANGED, "global", new_dir, FALSE, curbuf);

Advantages: 

- removes the need for multiple event types.
- future-proof for when tab-local directories might be added in the future (Neovim has that already)